fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/MiddlewareProvider.kt

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import kotlinx.coroutines.CoroutineScope
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckMiddleware
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Provides middleware for review quality check store.
+ */
+object MiddlewareProvider {
```

Comment:
Since this was inspired by `EngineMiddleware`, I am wondering if we should use a more specific naming such as `ReviewQualityMiddleware` or `ShoppingMiddleware` to avoid sounding too generic here and in the event that we add more `MiddlewareProvider` in other packages we run into the issue of finding the right provider.

Reply:
Yeah makes sense, renamed to `ReviewQualityCheckMiddlewareProvider` to be consistent. Also, moved to `di` package that brings in all the dependencies and Fragment doesn't depend on the `middleware` package